peps/pep-0501.rst

```diff
@@ -49,42 +75,50 @@ untrusted user: it's an opening for a form of code injection attack, where
 the supplied user data has not been properly escaped before being passed to
 the ``os.system`` call.
 
+While the `LiteralString` type annotation introduced in :pep:`675` means that typecheckers
+are able to report a type error for this kind of unsafe function usage, those errors don't
+help make it easier to write code that uses safer alternatives (such as `subprocess.run`).
+
 To address that problem (and a number of other concerns), this PEP proposes
 the complementary introduction of "t-strings" (a mnemonic for "template literal strings"),
-where ``f"Message with {data}"`` would produce the same
-result as ``format(t"Message with {data}")``.
-
-
-While this PEP and :pep:`675` are similar in their goals, neither one competes with the other,
-and can instead be used together.
-
-This PEP was previously in deferred status, pending further experience with :pep:`498`'s
-simpler approach of only supporting eager rendering without the additional
-complexity of also supporting deferred rendering. Since then, f-strings have become very popular
-and :pep:`701` has been introduced. This PEP has been updated to reflect current knowledge of f-strings,
-and improvements from 701. It is designed to be built on top of the :pep:`701` implementation.
+where ``format(t"Message with {data}")`` would produce the same result as
+``f"Message with {data}"``, but the interpolation template instance could instead be passed
+to other template rendering functions which processed the contents of the template
+differently.
 
 
 Proposal
 ========
 
+Dedicated template literal syntax
+---------------------------------
+
 This PEP proposes a new string prefix that declares the
 string to be a template literal rather than an ordinary string::
 
     template = t"Substitute {names:>10} and {expressions()} at runtime"
 
 This would be effectively interpreted as::
 
-    _raw_template = "Substitute {names:>10} and {expressions()} at runtime"
+    _raw_template = r"Substitute {names:>10} and {expressions()} at runtime"
     _parsed_template = (
-        ("Substitute ", "names"),
-        (" and ", "expressions()"),
-        (" at runtime", None),
+        (r"Substitute ", "names"),
+        (r" and ", "expressions()"),
+        (r" at runtime", None),
     )
     _field_values = (names, expressions())
     _format_specifiers = (f">10", f"")
+    _conversion_specifiers = ("", "")
```

Comment:
Building multiple parallel sequences like this is likely to be a pain to implement in the compiler, so I'm going to rework this to instead align with the way PEP 750 works (i.e. the compiler creates the template components and passes those to the TemplateLiteral constructor via `*args`)

Reply:
Essentially what https://peps.python.org/pep-0750/#evaluating-tag-strings describes, just with `types.TemplateLiteral` instead of `mytag`, and an extra leading arg for the full raw template string.